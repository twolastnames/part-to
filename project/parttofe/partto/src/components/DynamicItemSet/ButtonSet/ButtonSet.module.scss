.buttonSet {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  width: 40px;
  gap: 8px;
}
