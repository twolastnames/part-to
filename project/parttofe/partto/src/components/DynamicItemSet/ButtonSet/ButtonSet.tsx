import React from "react";

import classes from "./ButtonSet.module.scss";
import { Operation } from "../DynamicItemSet";
import { Button, ButtonProps } from "../../Button/Button";

export interface ButtonSetProps {
  operations: Array<ButtonProps>;
}

export function ButtonSet(props: ButtonSetProps) {
  return (
    <div className={classes.buttonSet} data-testid="ButtonSet">
      {props.operations.map((operation: Operation) => (
        <Button {...operation} />
      ))}
    </div>
  );
}
