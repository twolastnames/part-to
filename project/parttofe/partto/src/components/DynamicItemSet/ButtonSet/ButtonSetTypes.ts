import { ButtonProps } from "../../Button/ButtonTypes";

export interface ButtonSetProps {
  operations: Array<ButtonProps>;
}
