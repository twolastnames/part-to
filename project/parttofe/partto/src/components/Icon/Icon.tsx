import React, { ForwardRefExoticComponent, RefAttributes } from "react";
import { rem } from "@mantine/core";
import {
  Icon as TablerIcon,
  IconProps as TablerIconProps,
  IconPlayerTrackNext,
  IconPlayerPlay,
  IconList,
  IconFile,
  IconCaretDown,
  IconCaretUp,
  IconChefHat,
  IconCooker,
  IconToolsKitchen,
} from "@tabler/icons-react";

import classes from "./Icon.module.scss";

export type IconType = ForwardRefExoticComponent<
  TablerIconProps & RefAttributes<TablerIcon>
>;

export enum Size {
  Small = 20,
  Medium = 30,
  Large = 50,
  ExtraLarge = 80,
}

export interface IconProps {
  definition: IconType;
  size?: Size;
  onClick?: () => void;
}

export function Icon({ onClick, definition, size }: IconProps) {
  return React.createElement(definition, {
    ...(onClick
      ? {
          onClick,
          className: classes.clickable,
        }
      : {}),
    style: {
      width: rem((size || Size.Medium) as number),
      height: rem((size || Size.Medium) as number),
    },
  });
}

export const Next = IconPlayerTrackNext;
export const Start = IconPlayerPlay;
export const List = IconList;
export const File = IconFile;
export const Up = IconCaretUp;
export const Down = IconCaretDown;
export const ChefHat = IconChefHat;
export const Oven = IconCooker;
export const KitchenTools = IconToolsKitchen;
