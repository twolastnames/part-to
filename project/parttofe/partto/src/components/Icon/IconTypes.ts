import {
  Icon as TablerIcon,
  IconProps as TablerIconProps,
} from "@tabler/icons-react";
import { ForwardRefExoticComponent, RefAttributes } from "react";

export type IconType = ForwardRefExoticComponent<
  TablerIconProps & RefAttributes<TablerIcon>
>;

export enum Size {
  Small = 20,
  Medium = 30,
  Large = 50,
  ExtraLarge = 80,
}

export interface IconProps {
  definition: IconType;
  size?: Size;
  onClick?: () => void;
}
