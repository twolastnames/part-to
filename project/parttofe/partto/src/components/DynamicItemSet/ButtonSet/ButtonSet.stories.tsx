import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { ButtonSet } from "./ButtonSet";
import { ShellProvider } from "../../../ShellProvider";

const meta: Meta<typeof ButtonSet> = {
  component: ButtonSet,
  decorators: (Story) => (
    <ShellProvider>
      <Story />
    </ShellProvider>
  ),
};
export default meta;

type Story = StoryObj<typeof ButtonSet>;

export const Simple: Story = {
  args: {},
};
