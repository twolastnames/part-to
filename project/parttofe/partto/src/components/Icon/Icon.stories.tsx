import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { Icon, Next } from "./Icon";
import { ShellProvider } from "../../providers/ShellProvider";
import { Size } from "./IconTypes";

const meta: Meta<typeof Icon> = {
  component: Icon,
  decorators: (Story) => (
    <ShellProvider>
      <Story />
    </ShellProvider>
  ),
};
export default meta;

type Story = StoryObj<typeof Icon>;

export const Medium: Story = {
  args: { definition: Next },
};

export const Small: Story = {
  args: { definition: Next, size: Size.Small },
};

export const Large: Story = {
  args: { definition: Next, size: Size.Large },
};

export const ExtraLarge: Story = {
  args: { definition: Next, size: Size.ExtraLarge },
};
