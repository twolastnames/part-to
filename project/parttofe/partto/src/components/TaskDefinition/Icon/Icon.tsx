import React from "react";

import classes from "./Icon.module.scss";
import { ClassNames, IconProps } from "./IconTypes";
import {
  Duty as DutyIcon,
  Imminent as ImminentIcon,
  Icon as PhysicalIcon,
  Task as TaskIcon,
} from "../../Icon/Icon";

export const Duty: ClassNames = {
  imminent: classes.hidden,
  duty: classes.icon,
  task: classes.hidden,
};

export const Task: ClassNames = {
  imminent: classes.hidden,
  duty: classes.hidden,
  task: classes.icon,
};

export const Imminent: ClassNames = {
  imminent: classes.icon,
  duty: classes.hidden,
  task: classes.hidden,
};

export function Icon({ classNames: { task, duty, imminent } }: IconProps) {
  return (
    <>
      <span className={imminent}>
        <PhysicalIcon definition={ImminentIcon} />
      </span>
      <span className={duty}>
        <PhysicalIcon definition={DutyIcon} />
      </span>
      <span className={task}>
        <PhysicalIcon definition={TaskIcon} />
      </span>
    </>
  );
}
