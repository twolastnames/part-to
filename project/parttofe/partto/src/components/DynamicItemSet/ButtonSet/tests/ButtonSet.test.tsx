import React from "react";
import { expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { ButtonSet } from "../ButtonSet";
import { Start } from "../../../Icon/Icon";
import { ShellProvider } from "../../../../ShellProvider";

test("snapshot", () => {
  render(
    <ShellProvider>
      <ButtonSet
        operations={[
          {
            icon: Start,
            text: "start cooking",
            onClick: () => console.log("clicked"),
          },
        ]}
      />
    </ShellProvider>,
  );
  const component = screen.getByTestId("ButtonSet");
  expect(component).toMatchSnapshot();
});
