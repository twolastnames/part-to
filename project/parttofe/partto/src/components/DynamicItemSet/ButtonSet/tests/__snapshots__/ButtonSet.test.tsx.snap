// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="buttonSet"
  data-testid="ButtonSet"
>
  <div
    class="container"
  >
    <button
      class="mantine-focus-auto mantine-active button m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-testid="Button"
      style="--button-color: var(--mantine-color-white);"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        />
      </span>
    </button>
    <span
      class="icon"
    >
      <svg
        class="tabler-icon tabler-icon-player-play clickable"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 4v16l13 -8z"
        />
      </svg>
    </span>
  </div>
</div>
`;
